### PR TITLE
Document ability to add custom filesystems

### PR DESCRIPTION
**Pull request by brenelz**

### CODE REVIEW

File: filesystem.md [hunk 1]
```diff
@@ -106,3 +107,38 @@ The `Storage` facade may be used to interact with any of your configured disks.
...
+
+	use Illuminate\Contracts\Filesystem\Factory as FilesystemFactory;
+
+	class DropboxFilesystemServiceProvider {
+		
+		public function boot(FilesystemFactory $filesystem)
+		{
+			$filesystem->extend('dropbox', function($app, $config) 
+			{
+				$client = new Client($config['accessToken'], $config['clientIdentifier']);
```
Comment (GrahamCampbell) [inline filesystem.md]: You need to import this.

File: filesystem.md [hunk 2]
```diff
@@ -106,3 +107,38 @@ The `Storage` facade may be used to interact with any of your configured disks.
...
+				$adapter = new DropboxAdapter($client);
```
Comment (GrahamCampbell) [inline filesystem.md]: You need to import this.

File: filesystem.md [hunk 3]
```diff
@@ -106,3 +107,38 @@ The `Storage` facade may be used to interact with any of your configured disks.
...
+				
+				return new Filesystem($adapter);
```
Comment (GrahamCampbell) [inline filesystem.md]: You need to import this.

**Comment by brenelz** [comment]
Fully revised PR here - https://github.com/laravel/docs/pull/1140

**Pull request closed without merging**